### PR TITLE
WAZO-3572: Add test coverage for documentation generator

### PR DESCRIPTION
**Pull request by ja0x**

### CODE REVIEW

File: integration_tests/assets/docker-compose.documentation.override.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ services:
       - ../../xivo_bus:/app/xivo_bus:ro
 
   spec-validator:
-    image: asyncapi/cli:latest
+    image: asyncapi/cli
```
Comment (DrPyser) [inline integration_tests/assets/docker-compose.documentation.override.yml:17]: why remove the tag? Pretty sure no tag is equivalent to `latest`. You can also pin the version to the current one(`asyncapi/cli:1.2.34`)

File: integration_tests/Makefile [hunk 2]
```diff
@@ -1,8 +1,11 @@
-test-setup: egg-info bus-test bus-documentation
+test-setup: egg-info bus-test bus-documentation asyncapi-cli-pull
 
 test:
 	pytest -x
 
+asyncapi-cli-pull:
+	docker pull asyncapi/cli
```
Comment (DrPyser) [inline integration_tests/Makefile:7]: If you want the docker-compose file to rely only local images pulled from the Makefile, you could use a specific name when pulling. e.g.
```bash
docker pull asyncapi/cli:latest
docker tag asyncapi/cli:latest local/asyncapi/cli
```
then use `local/asyncapi/cli` in the compose file.

Comment (fblackburn1) [reply integration_tests/Makefile:7]: In the past, we removed all pulled logic from Makefile. wazo-test-helper already take care of it
For dev you can use `docker compose pull`

Comment (DrPyser) [reply integration_tests/Makefile:7]: You mean `wazo-test-helper` pulls images itself based on the docker-compose files?

Comment (ja0x) [reply integration_tests/Makefile:7]: I'll rewrite to use the `pull_containers` method instead and remove the one in the makefile.  Will be more in line with other wazo services

Comment (DrPyser) [reply integration_tests/Makefile:7]: Yep seems so. if we want want docker-compose to work without a make invocation then ignore my comment.

File: integration_tests/assets/docker-compose.documentation.override.yml (hunk 1 above)
Comment (ja0x) [reply integration_tests/assets/docker-compose.documentation.override.yml]: Readded the latest tag for now

File: contribs/documentation.py [hunk 3]
```diff
@@ -117,20 +125,24 @@ def recurse(
...
+                    if len(args) > 0:
+                        return recurse(type(subtype), enum=[subtype, *args])
+                    return recurse(type(subtype), const=subtype)
 
                 # i.e: Union[X, Y] or X | Y
+                # Note: we currently do not support multiple type in the API,
+                #       other types will be omitted from the spec
                 elif origin_type in (UnionType, Union):
                     nullable = any([item is NoneType for item in args])
-                    return recurse(subtype, *args, nullable=nullable)
```
Comment (ja0x) [inline contribs/documentation.py:125]: This is the actual fix to prevent the yaml to generate an invalid field in the spec when using unions with None

**Review by DrPyser — APPROVED** [review]
Seems good, works well.
For ref the diff with yesterday's versions of the generated asyncapi specs:
```diff
diff --git a/tmp/xivo_bus_asyncapi/wazo-plugind.yml b/tmp/xivo_bus_asyncapi_2/wazo-plugind.yml
index 59db3bf..30ebd53 100644
--- a/tmp/xivo_bus_asyncapi/wazo-plugind.yml
+++ b/tmp/xivo_bus_asyncapi_2/wazo-plugind.yml
@@ -42,9 +42,11 @@ channels:
                 type: string
               resource:
                 type: string
+                const: plugins
               details:
                 type: object
                 properties: {}
+            nullable: true
         headers:
           type: object
           properties:
@@ -87,9 +89,11 @@ channels:
                 type: string
               resource:
                 type: string
+                const: plugins
               details:
                 type: object
                 properties: {}
+            nullable: true
         headers:
           type: object
           properties:
diff --git a/tmp/xivo_bus_asyncapi/wazo-sysconfd.yml b/tmp/xivo_bus_asyncapi_2/wazo-sysconfd.yml
index ef52d99..e30a3af 100644
--- a/tmp/xivo_bus_asyncapi/wazo-sysconfd.yml
+++ b/tmp/xivo_bus_asyncapi_2/wazo-sysconfd.yml
@@ -39,6 +39,7 @@ channels:
             type: array
             items:
               type: string
+              format: uuid
         headers:
           type: object
           properties:
@@ -72,7 +73,7 @@ channels:
             format: uuid
           request_context:
             type: object
-            properties: !!python/name:builtins.NoneType ''
+            properties: {}
             nullable: true
           status:
             type: string

```
So this corrects the nullable support, and adds const attributes.

**Pull request merged**